ssl/statem/statem_clnt.c

```diff
@@ -1095,6 +1095,7 @@ WORK_STATE ossl_statem_client_post_process_message(SSL *s, WORK_STATE wst)
                  ERR_R_INTERNAL_ERROR);
         return WORK_ERROR;
 
+    case TLS_ST_CR_CERT_VRFY:
     case TLS_ST_CR_CERT_REQ:
         return tls_prepare_client_certificate(s, wst);
```

Comment:
So all that keeps us from calling tls_prepare_client_certificate() twice is the logic that is supposed to only return MSG_PROCESS_CONTINUE_PROCESSING once during any given handshake?

Reply:
Right - post processing is always an optional step that only ever happens if the main processing step decides we need to. Therefore if we've get this far then we must need to do it. It is up to the logic in the main processing steps to ensure that we only ever do it once.